### PR TITLE
Ort diff-tool for t3c 

### PR DESCRIPTION
**Pull request by jpappa200**
## What does this PR (Pull Request) do?
This is the diff-tool that will be used in the rewrite of ORT to go.

- [X] This PR is not related to any Issue that I know of. 

## Which Traffic Control components are affected by this PR?

- Traffic Ops ORT

## What is the best way to verify this PR?

To test this tool I compared files in `config_bkp/` to files in `config_trops/`
`go run t3c-diff-tool.go -d /path/to/config_bkp/<file> -t /path/to/config_trops/<file>`
## The following criteria are ALL met by this PR

- [X] This PR includes tests OR I have explained why tests are unnecessary
- [X] This PR includes documentation OR I have explained why documentation is unnecessary
- [X] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [X] This PR includes any and all required license headers
- [X] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_ops_ort/t3c-diff-tool/t3c-diff-tool.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+	"github.com/apache/trafficcontrol/traffic_ops_ort/t3cutil"
+	"github.com/kylelemons/godebug/diff"
+	"github.com/pborman/getopt/v2"
+)
+
+func main() {
+	different := false
+	tropsFile := getopt.StringLong("trops-file", 't', "", "Required: Config file name in Traffic Ops")
+	diskFile := getopt.StringLong("disk-file", 'd', "", "Required: Config file on disk")
```
Comment (ocket8888) [inline traffic_ops_ort/t3c-diff-tool/t3c-diff-tool.go:36]: If you make these true [Mandatory Options](https://godoc.org/github.com/pborman/getopt/v2#hdr-MANDATORY_OPTIONS), you won't need to check them yourself.

File: traffic_ops_ort/t3c-diff-tool/t3c-diff-tool.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ */
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/apache/trafficcontrol/traffic_ops_ort/t3cutil"
+	"github.com/kylelemons/godebug/diff"
```
Comment (ocket8888) [inline traffic_ops_ort/t3c-diff-tool/t3c-diff-tool.go:29]: New dependencies must be vendored

File: traffic_ops_ort/t3cutil/t3cutil.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		s = el.ReplaceAllString(s, "")
+		s = am.ReplaceAllString(s, "")
+		s = lt.ReplaceAllString(s, "<")
+		s = gt.ReplaceAllString(s, ">")
+		s = strings.TrimSpace(s)
+		newlines = append(newlines, s)
+	}
+
+	return newlines
+}
```
Comment (ocket8888) [inline traffic_ops_ort/t3cutil/t3cutil.go:91]: This is available as [a function in the Go standard library](https://godoc.org/html#UnescapeString)

File: traffic_ops_ort/t3cutil/t3cutil.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	"strings"
+)
+
+// commentsFilter is used to remove comment
+// lines from config files while making
+// comparisons.
+func CommentsFilter(body []string) []string {
+	var newlines []string
+
+	newlines = make([]string, 0)
```
Comment (ocket8888) [inline traffic_ops_ort/t3cutil/t3cutil.go:36]: If you're going to use `make`, you could pass it a third parameter to pre-allocate space, which would speed things up.

**Review by ocket8888 — COMMENTED** [review]
This could use a test or two

**Review by rob05c — APPROVED** [review]
Looks good! An important piece of the ORT UNIX-style rewrite.
We could nit-pick it, e.g. returning nothing on success, error codes, etc.
But it works, code looks good, I don't see any bugs.

I tested, it works as expected. Prints the diff if there is any, `false` if not`.

It's experimental, doesn't need a changelog or docs. And it needs an Integration Test Framework, which is still being developed.

I'm +1 on merging this in, and we can iteratively decide the exact interface etc later.

**Review by ocket8888 — APPROVED** [review]
Appears to work well.

**Pull request merged**